src/java/org/apache/cassandra/utils/SyncUtil.java

```diff
@@ -149,18 +106,8 @@ public static void sync(FileDescriptor fd) throws SyncFailedException
                 throw new RuntimeException(e);
             }
 
-            int useCount = 1;
-            try
-            {
-                if (fdUseCountField != null)
-                    useCount = ((AtomicInteger)fdUseCountField.get(fd)).get();
-            }
-            catch (Exception e)
-            {
-                throw new RuntimeException(e);
-            }
-            if (closed || !fd.valid() || useCount < 0)
-                throw new SyncFailedException("Closed " + closed + " valid " + fd.valid() + " useCount " + useCount);
+            if (closed || !fd.valid())
```

Comment:
This "sync" method is unused and can be removed entirely. Then, we can drop all the reflections from this class.